docs/internal-documentation/maintenance/SystemTesting.md

```diff
@@ -0,0 +1,200 @@
+System testing
+==============
+Before a new version of Zonemaster can be released it must pass three levels of
+system testing: Installation Testing, Smoke Testing and Acceptance Testing.
+Each test level is described in a separate section below. A separate section is
+reserved for the concept of Configurations, which is used by other sections.
+
+
+1. Configurations
+-----------------
+A configuration is a tuple of the following parameters:
+
+* architecture (e.g. x86_64)
+* operating system version (e.g. Ubuntu 14.04)
+* locale (e.g. en_US.UTF-8)
+* database engine version (e.g. PostgreSQL 9.3)
+
+Each parameter has a section below definig the parameters more closely.
+
+The complete configuration space consists of all combinations of the
+parameters. Each test level section specifies a part of the configuration space
+to test.
+
+
+### Architecture
+
+The only architecture we support is amd64/x86_64.
+
+
+### Operating system version
+
+The following operating systems are supported:
+
+* Centos
+* Debian
+* FreeBSD
+* Ubuntu
+
+For each supported operating system, the supported versions are determined using
+the following criteria.
+
+* CentOS:
+  * The supported versions are the Base Distributions (excluding 5 and 6) as
+    specified at: 
+    https://en.wikipedia.org/wiki/CentOS#End-of-support_schedule
+  * The supported minor release is the one specified at:
+    https://wiki.centos.org/Download
+
+* Debian:
+  * The supported versions are "stable" and "oldstable" as specified at: 
+    https://wiki.debian.org/DebianReleases#Current_Releases.2FRepositories
+  * The supported point release is the current one as specified at:
+    https://wiki.debian.org/DebianReleases/PointReleases
+
+* FreeBSD:
+  * The supported versions are the Extended releases (excluding 9.3) as
+    specified at: https://www.freebsd.org/security/
+  * Patch level should not be specified.
+
+* Ubuntu:
+  * The supported versions are the LTS releases (excluding 12.04) as specified
+    at: https://wiki.ubuntu.com/Releases
+  * The supported patch level is the one specified at:
+    https://wiki.ubuntu.com/Releases
+
+
+### Locale
+
+The "C" locale is supported along with the locales supported for language
+translation.
```

Comment:
The correct syntax for fr_??.utf8 is `fr.UTF-8`.
The runtime behaviour of CLI should be affected by the system locale. That should be reflected in test procedures rather than the test configurations.
The runtime behaviour of backend and GUI should not be affected by their respective system locales. There should be test configurations to cover these.